next/lib/utils/themeContext.tsx

```diff
@@ -1,3 +1,4 @@
+'use client';
```

Comment:
We should no longer use createContext or generally wrapper components with `"use client"` declaration. In this case, all components within the provider (below a section) would also be client components. See https://nextjs.org/docs/app/building-your-application/rendering/client-components#using-client-components-in-nextjs

> [...] This means that by defining a "use client" in a file, all other modules imported into it, including child components, are considered part of the client bundle.

This could be solved by an inherited fusion context variable, but that doesn't work in the backend (at least in my tests).

Reply:
I understand it the way, that children passed via props can be server components and are not part of the client bundle (see https://nextjs.org/docs/app/building-your-application/rendering/composition-patterns#interleaving-server-and-client-components). The note in the doc should only apply to imported child components AFAIK.

So AFAIK it's not really a problem but we should check that it really works that way.

Reply:
You're right! I've tested it 👍🏼